test/commands/stop.js

```diff
@@ -11,22 +11,183 @@
+require('superagent-proxy')(request);
+const collectorName = 'collector1';
+const refocusUrl = 'http://www.refocusserver.com';
+const accessToken = 'abcdefghijklmnopqrstuvwxyz';
+const collectorToken = 'zyxwvutsrqponmlkjihgfedcba';
+const refocusProxy = 'http://refocusproxy.com';
+const dataSourceProxy = 'http://datasourceproxy.com';
+const httpStatus = require('../../src/constants.js').httpStatus;
 
-describe('test/commands/stop >', () => {
+beforeEach((done) => {
+  nock(refocusUrl, {
+    reqheaders: { authorization: accessToken },
+  })
+  .post('/v1/collectors/start',
+    { name: `${collectorName}`, version: '1.0.0' })
+  .reply(httpStatus.CREATED, { token: collectorToken });
+
+  nock(refocusUrl, {
+    reqheaders: { authorization: accessToken },
+  })
+  .post(`/v1/${collectorName}/stop`, { name: `${collectorName}` })
+  .reply(httpStatus.OK);
+  done();
+});
 
+describe('test/commands/stop >', () => {
   // TODO: child process fails on travis with error
   // /bin/sh: 1: refocus-collector: not found
-  it('logs the expected result', (done) => {
+  it('not ok, cannot run stop without running start first', (done) => {
     const { exec } = require('child_process');
-    exec('refocus-collector stop --collectorName=PRD_Collector_12345',
-      (error, stdout, stderr) => {
+    exec('refocus-collector stop',
+    (error, stdout, stderr) => {
+      console.log('error--', error);
       if (error) {
         console.error(`exec error: ${error}`);
-        done(error);
+        return done();
       }
 
-      expect(stdout).to.contain('Stop => ');
+      return done('Expected Stop to fail');
+    });
+  });
+
+  it('stop without proxy', (done) => {
+    start.execute(collectorName, refocusUrl, accessToken, {})
+    .then(() => {
+      // make sure start created the repeat
+      expect(repeater.tracker.heartbeat).to.be.an('object');
+    })
+    .then(() => stop.execute())
+    .then((res) => {
+      expect(res.status).to.equal(httpStatus.OK);
+      expect(repeater.tracker.heartbeat).to.equal(undefined);
+      repeater.stopAllRepeat();
       done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('stop with proxy', (done) => {
+    start.execute(collectorName, refocusUrl, accessToken,
+      { refocusProxy, dataSourceProxy }
+    )
+    .then(() => {
+      // make sure start created the repeat
+      expect(repeater.tracker.heartbeat).to.be.an('object');
+    })
+    .then(() => stop.execute(true))
+    .then((res) => {
+      expect(res.status).to.equal(httpStatus.OK);
+      expect(repeater.tracker.heartbeat).to.equal(undefined);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('force stop, flush should not be called', (done) => {
+    start.execute(collectorName, refocusUrl, accessToken,
+      { refocusProxy, dataSourceProxy }
+    )
+    .then(() => {
+      // make sure start created the repeat
+      expect(repeater.tracker.heartbeat).to.be.an('object');
+    })
+    .then(() => stop.execute())
```

Comment:
It should be stop.execute(true) if this is testing force stop, right?
And is there a way to actually test that the queues aren't flushed?

Reply:
yes, will fix it.